certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/service/GetCertificateEventsOfTypeDomainService.java

```diff
@@ -0,0 +1,146 @@
+package se.inera.intyg.certificateservice.domain.certificate.service;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import se.inera.intyg.certificateservice.domain.certificate.model.Certificate;
+import se.inera.intyg.certificateservice.domain.certificate.model.CertificateEvent;
+import se.inera.intyg.certificateservice.domain.certificate.model.CertificateEventType;
+import se.inera.intyg.certificateservice.domain.certificate.model.RelationType;
+import se.inera.intyg.certificateservice.domain.message.model.MessageType;
+
+public class GetCertificateEventsOfTypeDomainService {
+
+  public List<CertificateEvent> events(Certificate certificate, CertificateEventType type) {
+    switch (type) {
+      case CREATED -> {
+        return List.of(eventFromTimestamp(certificate, certificate.created(), type));
+      }
+      case SIGNED, AVAILABLE_FOR_PATIENT -> {
+        return List.of(eventFromTimestamp(certificate, certificate.signed(), type));
+      }
+      case SENT -> {
+        return certificate.sent() != null
+            ? List.of(eventFromTimestamp(certificate, certificate.sent().sentAt(), type))
+            : Collections.emptyList();
+      }
+      case INCOMING_MESSAGE -> {
+        return certificate.messages()
+            .stream()
+            .filter(message -> message.type() != MessageType.ANSWER
```

Comment:
Feels a bit overkill since it's just checking one value. If two types matched to the same kind I think it would add value. I looked and we've only done it on draft before which I'm guessing is because draft will have several statuses when we introduce locked. I'll leave this as it is.